src/test/clojure/clara/test_rules.clj

```diff
@@ -5452,5 +5452,25 @@
                     (insert (->Cold -10))
                     (fire-rules {:cancelling true})
                     (query cold-query))))))
-    
-  
+
```

Comment:
I'm adding a test with the following structure, which I think is what you're looking for.

```clj
        check-reuse-previous-binding (dsl/parse-query [] [[WindSpeed (= ?t windspeed) (= ?w windspeed)]
                                                          [Temperature (= ?t temperature)
                                                           (join-filter-equals ?w ?t 10)]])
```